Docs: Improve Conda docs - PyPI + lock files (#5531)
nextflow-io/nextflow: docs/conda.md

The environment file name **must** have a `.yml` or `.yaml` extension or else it won't be properly recognised.
:::

(conda-pypi)=
### Python Packages from PyPI

> Conda environment files can also be used to install Python packages from the [PyPI repository](https://pypi.org/)), through the `pip` package manager (which must also be explicitly listed as a required package):

Reviewer: ```suggestion
Conda environment files can also be used to install Python packages from the [PyPI repository](https://pypi.org/), through the `pip` package manager (which must also be explicitly listed as a required package):
```

```yaml
name: my-env-2
channels:
  - defaults
dependencies:
  - pip
  - pip:
    - numpy
    - pandas
    - matplotlib
```

### Conda text files

> It is possible to provide the dependencies using a plain text file, just listing each package name as a separate line. For example:

Reviewer 2: ```suggestion
It is possible to provide dependencies by listing each package name as a separate line in a plain text file. For example:
```

Like before, the extension matters. Make sure the dependencies file has a `.txt` extension.
:::

### Conda lock files

The final way to provide packages to Conda is with [Conda lock files](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#identical-conda-envs).

> These are generated from existing conda environements using the following command:

Reviewer 2: ```suggestion
These are generated from existing Conda environments using the following command:
```

```bash
conda list --explicit > spec-file.txt
```

or if using Mamba / Micromamba:

```bash
micromamba env export --explicit > spec-file.txt
```

> They can also be downloaded from Wave build pages.

Reviewer 2: ```suggestion
Conda lock files can also be downloaded from Wave build pages.
```

> Conda lock files can also be downloaded from Wave build pages.

Reviewer: ```suggestion
Conda lock files can also be downloaded from [Wave](https://seqera.io/wave/) build pages.
```

> These files include the every package, including dependencies. As such, no conda environment resolution step is needed. This is faster and also better for reproducibility between runs.

Reviewer 2: ```suggestion
These files include every package and their dependencies. As such, no Conda environment resolution step is needed. This is faster and more reproducible.
```

> The files contain package URLs and optionally also a md5hash for each download to confirm identity:

Reviewer 2: ```suggestion
The files contain package URLs and an optional md5hash for each download to confirm identity:
```
